parts/linux/cloud-init/artifacts/ubuntu/cse_helpers_ubuntu.sh

```diff
@@ -40,22 +43,38 @@ apt_get_update() {
     echo Executed apt-get update $i times
     wait_for_apt_locks
 }
-apt_get_install() {
-    retries=$1; wait_sleep=$2; timeout=$3; shift && shift && shift
+apt_get_update() {
+    _apt_get_update 10 ""
+}
+_apt_get_install() {
+    local retries=$1
+    local wait_sleep=$2
+    local apt_opts=$3
+    shift && shift && shift
+    local packages=("$@")
+
     for i in $(seq 1 $retries); do
         wait_for_apt_locks
         export DEBIAN_FRONTEND=noninteractive
         dpkg --configure -a --force-confdef
-        apt-get install -o Dpkg::Options::="--force-confold" --no-install-recommends -y ${@} && break || \
+
+        if apt-get install ${apt_opts} -y "${packages[@]}"; then
+            echo "Executed apt-get install \"${packages[@]}\" $i times"
+            wait_for_apt_locks
+            return 0
+        fi
+
         if [ $i -eq $retries ]; then
             return 1
         else
             sleep $wait_sleep
             apt_get_update
         fi
     done
-    echo Executed apt-get install --no-install-recommends -y \"$@\" $i times;
-    wait_for_apt_locks
+}
+apt_get_install() {
+    retries=$1; wait_sleep=$2; timeout=$3; shift && shift && shift
+    _apt_get_install $retries $wait_sleep '--no-install-recommends -o Dpkg::Options::="--force-confold"' "$@"
```

Comment:
it eventually runs `apt-get install --no-install-recommends -o 'Dpkg::Options::="--force-confold"' -y apparmor-utils`
will fail with error dpkg: error: need an action option

there should be no queto before Dpkg